lab_2/main.py

```diff
@@ -0,0 +1,232 @@
+        if (pSearch):
+            P.append(Input[i])
+    return F, P
+
+
+def FP_Pairing(F, P):
+    Frow = len(F)
+    Prow = len(P)
+    cols = len(F[0])
+    pairs = list()
+    noMatch = [u'Unmatched with corresponding PP']
+    if (len(F[0]) != len(P[0])):
+        print
+        'Error, inputs have different numnber of columns'
+    for i in range(Frow):
+        matched = 0
+        name = F[i][0][:-2]
+        pat = re.compile('^' + re.escape(name))
+        for j in range(Prow):
+            match = pat.search(P[j][0])
+            if match:
+                pairs.append([F[i], P[j]])
+                matched = 1
+        if (not matched):
+            noMatch.append(F[i][0])
+    return pairs, noMatch
+
+
+def Nefa_Sort(pairs):
+    ne = []
+    to = []
+    for i in pairs:
+        isNe = re.search('-NE-', i[0][0])
+        if isNe:
+            ne.append(i)
+        else:
+            to.append(i)
+    return ne, to
+
+
+def Org_Sum(pairs, orgType):
+    orgSum = dict()
+    orgTemp = dict()
+    hasNone = [u'Contains empty values']
+    rows = len(pairs)
+
+    if len(orgType) == 2:
+        for m in orgType[0]:
+            for k in orgType[1]:
+                orgSum[m + ': ' + k] = [[0] * 10, [0] * 10]
+                for j in range(rows):
+                    row = pairs[j]
+                    if (pairs[j][0][2] == m or pairs[j][0][1] == m) and not contains_Nonetype(row[0], row[1]) and (
+                            pairs[j][0][2] == k or pairs[j][0][1] == k):
+                        x = [orgSum[m + ': ' + k][0][z - 3] + row[0][z] for z in range(3, 13)]  # Fasting
+                        orgSum[m + ': ' + k][0] = x
+                        y = [orgSum[m + ': ' + k][1][z - 3] + row[1][z] for z in range(3, 13)]  # Post Prandial
+                        orgSum[m + ': ' + k][1] = y
+                    if (contains_Nonetype(row[0], row[1])):  # keeps log of incomplete data oxylipins
+                        hasNone.append(row[0][0][:-2])
+
+    else:
+        for m in orgType:
+            orgSum[m] = [[0] * 10, [0] * 10]
+            for j in range(rows):
+                row = pairs[j]
+                if (pairs[j][0][2] == m or pairs[j][0][1] == m) and not contains_Nonetype(row[0], row[1]):
+                    x = [orgSum[m][0][z - 3] + row[0][z] for z in range(3, 13)]  # Fasting
+                    orgSum[m][0] = x
+                    y = [orgSum[m][1][z - 3] + row[1][z] for z in range(3, 13)]  # Post Prandial
+                    orgSum[m][1] = y
+
+                if (contains_Nonetype(row[0], row[1])):  # keeps log of incomplete data oxylipins
+                    hasNone.append(row[0][0][:-2])
+
+    return orgSum, list(set(hasNone))
+
+
+def contains_Nonetype(in1, in2):
+    none = False
+    for x in in1:
+        if x is None:
+            none = True
+            return none
+    for y in in2:
+        if y is None:
+            none = True
+            return none
+    return none
```

Comment:
довольно громоздкая запись  с использованием двух циклов
альтернатива:
```suggestion
       return (None in data1) or (None in data2)
```